src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/Parsers/SearchValueExpressionBuilderHelper.cs

```diff
@@ -20,6 +20,7 @@ internal class SearchValueExpressionBuilderHelper : ISearchValueVisitor
 
         private string _searchParameterName;
         private SearchModifierCode? _modifier;
+        private string _targetTypeModifier;
```

Comment:
>_targetTypeModifier [](start = 23, length = 19)

I'm not quite understand why we need this field. All we do is assign value to it, but I can't find code which would read it.

Reply:
My thinking was: `_modifier` does not represent the complete state, so we would need this field if we wanted to handle modifier in some way. But I guess it could be added later if we need it. What do you think?

Reply:
Maybe we can alter `void ISearchValueVisitor.Visit(ReferenceSearchValue reference)` method to check what `reference.ResourceType` is equal to `_targetTypeModifier` to be more safe...

Reply:
It almost feels like SearchModifierCode should be inside a new container class or tuple, i.e. `SearchModifier(SearchModifierCode, ResourceType = null)`, thoughts?

Reply:
> It almost feels like SearchModifierCode should be inside a new container class or tuple, i.e. `SearchModifier(SearchModifierCode, ResourceType = null)`, thoughts?

Good idea! Will add a class